onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -172,7 +172,7 @@ Status TreeEnsembleCommon<InputType, ThresholdType, OutputType>::Init(
               nodes_falsenodeids.size() == nodes_values_as_tensor.size());
   ORT_ENFORCE(target_class_ids.size() == target_class_nodeids.size());
   ORT_ENFORCE(target_class_ids.size() == target_class_treeids.size());
-  ORT_ENFORCE(target_class_ids.size() == target_class_treeids.size());
+  ORT_ENFORCE(target_class_ids.size() == target_class_weights.size());
```

Comment:
Per line 179, it appears `target_class_weights` is allowed to be empty.

```
  ORT_ENFORCE(target_class_weights.empty() || target_class_weights_as_tensor.empty());
```

Could this new enforcement regress something else?

Should it be:

```
  ORT_ENFORCE(target_class_ids.size() == target_class_weights.size() || target_class_weights.empty());
```